### ISSUE TITLE
Build settings is not working

### ISSUE THREAD

**Issue by paololazzari** [RyanTheAllmighty/Docker-Hub-API#29]
`dockerHubAPI.buildSettings(username, repository)` is not working. It always returns 

> (node:36992) UnhandledPromiseRejectionWarning: Error: {"detail":"Object not found"}

**Comment by RyanTheAllmighty** [comment]
Looks like they may not have this endpoint anymore.

This library is quite out of date, it's been 5 years since I made this library and likely a bunch of the calls no longer work as Docker hub change their api over the years, especially since this is using undocumented api's.

Looks related to #25

**Comment by paololazzari** [comment]
Yeah, I figured that might have been the case. I am trying to add new build rules programmatically but I am not having much luck.
Do you have any tips/pointers on how that could be done for the new API?

I went through the docs here https://docs.docker.com/registry/spec/api/ but there seems to be no mention whatsoever on how that could be done

Thanks

**Comment by aideslucas** [comment]
@LazzariPaolo @RyanTheAllmighty  any progression regarding this?
I found from hub.docker.com the api:
https://hub.docker.com/api/build/v1/{USERNAME}/source/?image={USERNAME}%2F{REPONAME}
but if I try to send the same request with the JWT token I get 401 Unauthorized..
